server/datastore/mysql/apple_mdm.go

```diff
@@ -2862,6 +2862,65 @@ func (ds *Datastore) GetMDMAppleDefaultSetupAssistant(ctx context.Context, teamI
 	return asst.ProfileUUID, asst.UploadedAt, nil
 }
 
+func (ds *Datastore) UpdateHostDEPAssignProfileResponses(ctx context.Context, payload *godep.ProfileResponse) error {
+	if payload == nil {
+		// caller should not ensure this does not happen
+		level.Debug(ds.logger).Log("msg", "update host dep assign profiles responses received nil payload")
+		return nil
+	}
+
+	p := fleet.ToHostDEPAssignProfileResponses(payload)
+	if len(p.Failed)+len(p.NotAccessible)+len(p.Success) != len(payload.Devices) {
+		// this should never happen unless Apple changes the response format, so we log it for future debugging
+		level.Debug(ds.logger).Log("msg", "unrecognized assign profile response for one or more devices", "devices", payload.Devices)
+		return nil
+	}
```

Comment:
should we store whatever we get instead of trying to categorize this response?

Reply:
I wouldn't say that we're really categorizing the response rather just reducing the dictionary from Apple to three lists based on the reported status, which makes it a little easier to do the DB writes. We could just inline the `ToHostDEPAssignProfileResponses` here instead of having it in the `fleet` package if that makes it clearer why we're doing it.

Reply:
> This could very well be a single batch-insert where we build the params for each device in a loop (like we do in other places)

This needs to be an update though, so I'm not sure which example you had in mind? To batch the updates here, I loop over the Apple response to group into lists of serials that I can use for the `WHERE` clause.

Reply:
(unless we decide to add the serial number as the primary key of `host_dep_assignments`?)